keps/sig-node/3317-user-namespaces/README.md

```diff
@@ -0,0 +1,1015 @@
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+Some definitions first:
+- New kubelet: kubelet with CRI proto files that includes the changes proposed in
+this KEP.
+
+- Old kubelet: idem, but CRI proto files don't include this changes.
+
+- New runtime: container runtime with CRI proto files that includes the changes
+proposed in this KEP.
+
+- Old runtime: idem, but CRI proto files don't include this changes.
+
+New runtime and old kubelet: all works just fine. Kubelet doesn't request userns
+(doesn't have that feature) and therefore the runtime doesn't create them. The
+runtime can detect this situation as the `user` field in the `NamespaceOption`
+will be seen as nil, [thanks to protobuf][proto3-defaults]. We already tested
+this with real code.
+
+Old runtime and new kubelet: all will work fine. As the `user` field of the
+`NamespaceOption` message is not part of the runtime protofiles, that part is
+ignored by the runtime and pods are created using the host userns.
+
+
+[proto3-defaults]: https://developers.google.com/protocol-buffers/docs/proto3#default
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: UserNamespacesSupport
+  - Components depending on the feature gate: kubelet
```

Comment:
At least kube-apiserver too, as the API field will be gated by by this feature gate.

Reply:
Great catch, thanks! Updated